### PR TITLE
removed logic setting up default workers per model while fetching init workers for starting torchserve

### LINKED ISSUE #511: Model registered without any initial workers creates workers equal to available GPUs or CPU cores
## Context
This bug got introduced to PR #318 

* torchserve version: latest master
* torch version: 1.5
* torchvision version [if any]: NA
* torchtext version [if any]: NA
* torchaudio version [if any]: NA
* java version: JDK 11
* Operating System and version: Ubuntu 18.04/MacOS

## Your Environment

* Installed using source? [yes/no]: Yes
* Are you planning to deploy it using docker container? [yes/no]: No
* Is it a CPU or GPU environment?: Both
* Using a default/custom handler? [If possible upload/share custom handler/model]: No
* What kind of model is it e.g. vision, text, audio?: Any
* Are you planning to use local models from model-store or public url being used e.g. from S3 bucket etc.?
  [If public url then provide link.]: NA
* Provide config.properties, logs [ts.log] and parameters used for model registration/update APIs: 
```
curl -X POST  "http://localhost:8081/models?url=resnet-18.mar&model_name=resnet18"
```
The model was already available in the model store.

* Link to your project [if any]:NA

## Expected Behavior
`TorchServe` should not create any workers if no `initial_workers` are specified in the model registration request. Unless `default_num_workers` is configured to some specific value.

## Current Behavior
If `default_num_workers`  is not configured and `initial_workers` parameter is not supplied in the registration request, `TorchServe` should register the model with 0 workers.

## Possible Solution
Do not set `default_num_workers` to available GPUs or CPU cores in [configmanager.java]https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java#L312

## Steps to Reproduce
Register a model with no initial workers
```
curl -X POST  "http://localhost:8081/models?url=resnet-18.mar&model_name=resnet18"
```

## Failure Logs [if any]
```
curl -X POST  "http://localhost:8081/models?url=resnet-18.mar&model_name=resnet18"
{
  "status": "Model \"resnet18\" Version: 1.0 registered with 8 initial workers"
}
```

### PR DESCRIPTION
**Pull request by harshbafna**
## Description

 - The default number of workers was getting set to the number of GPUs/CPU cores.
 - This should be ideally set to 0 in case not configured through `config.properties`

Fixes #511

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Feature/Issue validation/testing

ubuntu@ip-172-31-54-27:~$ curl -X POST "http://localhost:8081/models?url=https://torchserve.s3.amazonaws.com/mar_files/resnet-18.mar&model_name=resnet18&synchronous=false"
{
  "status": "Model \"resnet18\" Version: 1.0 registered with 0 initial workers. Use scale workers API to add workers for the model."
}

[sanity_logs_511.txt](https://github.com/pytorch/serve/files/4898408/sanity_logs_511.txt)

## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java [hunk 1]
```diff
@@ -157,6 +157,21 @@ public void testDescribeApi() throws InterruptedException {
...
+        TestUtils.getLatch().await();
+        DescribeModelResponse[] resp =
+                JsonUtils.GSON.fromJson(TestUtils.getResult(), DescribeModelResponse[].class);
+        Assert.assertEquals(TestUtils.getHttpStatus(), HttpResponseStatus.OK);
+        Assert.assertEquals(resp[0].getMinWorkers(), configManager.getDefaultWorkers());
+    }
+
+    @Test(
+            alwaysRun = true,
+            dependsOnMethods = {"testInitialWorkers"})
```
Comment (dhaniram-kshirsagar) [inline frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java:174]: Add UT to validate the workers for model registered via api [with/without initial workers.]

Comment (harshbafna) [reply frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java:174]: Done.

**Comment by harshbafna** [comment]
> @harshbafna What will be the implications for the `load_models=all` scenario with this change? How will the model workers and other configs get set for that scenario -- which is generally used for Orchestration sol likes Sagemaker, Kubernetes

@chauhang 

For all the models loaded at the time of `TorchServe` startup, the initial number of workers will be selected as follows 

 - `default_workers_per_model` configured through `config.properties`
 - if not configured then it will use available number of GPUs/vCPU cores.

For models registered through API, if `initial_workers` parameter is not supplied then the number of workers will be  selected as follows :

 - `default_workers_per_model` configured through `config.properties`
 - if not configured then it will be set to 0. [This is behavior which got broken and is fixed by this PR]

**Comment by dhanainme** [comment]
Merged as https://github.com/pytorch/serve/pull/549

**Pull request closed without merging**